### PR TITLE
chore: Drop python 3.8 support and improve ci file name

### PR DESCRIPTION
**Pull request by Wh1isper**
## Description

Since python 3.8 has reached the end of its life cycle, we should drop its support.

## Types of changes

- [ ] Maintenance (no change in code, maintain the project's CI, docs, etc.)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/extension.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.11"]
+        python-version: ["3.9", "3.11", "3.12"]
```
Comment (jalr4ever) [inline .github/workflows/extension.yml:15]: Little confused here, why we just skip `3.10` version here?  See `python-package` contained `3.10` though.

Comment (Wh1isper) [reply .github/workflows/extension.yml]: My intention was to wait until python 3.12 was stable to test with just 3.9 and 3.12, and so far 3.12 seems to be stable enough. Switching to 3.12

Comment (jalr4ever) [reply .github/workflows/extension.yml]: Yeah. I might not have been very clear. Here I am actually confused about the `matrix`.

As you assigned it  ` python-version: ["3.9", "3.12"]` in `ci-test-extension.yml` , but why its matirx value `["3.9", "3.10", "3.11", "3.12"]` in `ci-test-python-package.yml`?

Should we assigned `["3.9", "3.10", "3.11", "3.12"]` value in `ci-test-extension.yml`?

Comment (Wh1isper) [reply .github/workflows/extension.yml]: There's no need for that, for the plugin there's no computational library dependency so there's basically no python version difference, we just need to make sure it runs on the oldest and newest (supproted) versions. For the core library, I'd like to make sure he runs on all supported python versions.

**Pull request merged**